packages/client/tests/e2e/custom-output-ts/tests/main.test.ts

```diff
@@ -0,0 +1,96 @@
+import path from 'path'
+import * as ts from 'typescript'
+
+const baseTsConfig = require('../../tsconfig.base.json').compilerOptions
+
+const mainTs = path.resolve(__dirname, '..', 'src', 'index.ts')
+const prismaClientName = ts.escapeLeadingUnderscores('PrismaClient')
+
+const deprecatedOptions = [
+  { module: 'commonjs', moduleResolution: 'Node' },
+  { module: 'ES2015', moduleResolution: 'Node' },
+  { module: 'ES2020', moduleResolution: 'Node' },
+  { module: 'ES2022', moduleResolution: 'Node' },
+  { module: 'ESNext', moduleResolution: 'Node' },
+  { module: 'Node16', moduleResolution: 'Node' },
+  { module: 'NodeNext', moduleResolution: 'Node' },
```

Comment:
These two point out that we maybe need to adjust our warning a little bit: changing just `module` without `moduleResultion` does not actually fixes the problem.

Reply:
Actually, TS API seems to be more forgiving than `tsc`, these options are invalid when checked through `tsc. Should we remove them?

Reply:
Yes, if we can improve our error message I am all for it, how would you change it?

Reply:
And yes, if the option is invalid then I guess we should remove it.

Reply:
Warning suggestion was coming from me not getting validation error from invalid combinations and thinking they are valid. Since then I figured out how to make TS API throw errors on those, our present warning is correct.